pkgs/development/compilers/solc/default.nix

```diff
@@ -42,13 +40,33 @@ stdenv.mkDerivation {
     "-DUSE_Z3=OFF"
   ];
 
-  doCheck = stdenv.hostPlatform.isLinux && stdenv.hostPlatform == stdenv.buildPlatform;
-  checkPhase = "LD_LIBRARY_PATH=./libsolc:./libsolidity:./libevmasm:./libdevcore:./libyul:./liblangutil:./test/tools/yulInterpreter:$LD_LIBRARY_PATH " +
-               "./test/soltest -p true -- --no-ipc --no-smt --testpath ../test";
-
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost ]
-    ++ stdenv.lib.optionals z3Support [ z3 ];
+  buildInputs = [ boost ] ++ stdenv.lib.optionals z3Support [ z3 ];
+  checkInputs = [ ncurses python2 ];
+
+  # Test fails on darwin for unclear reason
+  doCheck = stdenv.hostPlatform.isLinux;
+
+  checkPhase = ''
+    for dir in $(find . -type d);
+    do
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/$dir
+    done
+    pushd ..
+    # IPC tests need aleth avaliable, so we disable it
+    sed -i "s/IPC_ENABLED=true/IPC_ENABLED=false\nIPC_FLAGS=\"--no-ipc\"/" ./scripts/tests.sh
+    for i in ./scripts/*.sh; do
+      patchShebangs $i
+    done
+    for i in ./scripts/*.py; do
+      patchShebangs $i
+    done
+    for i in ./test/*.sh; do
+      patchShebangs $i
+    done
+    TERM=xterm ./scripts/tests.sh
+    popd
```

Comment:
Is this necessary?

Either way, this raises SC2164, although not sure how important it is for us.

Reply:
We always `pushd ..`, so it is probably fine